byway/src/main/webapp/recommendations.js

```diff
@@ -68,12 +73,7 @@ function addMainStopsToRegions() {
 /* Resets the alerts found in a previous attempt to load recommendations and reveals loading bar. */
 function resetUserAlerts() {
   document.getElementById("message-container").style.visibility = 'hidden';
-  document.getElementById("general-message").style.visibility = 'hidden';
-  let statusesContainer = document.getElementById("statuses");
-  while(statusesContainer.hasChildNodes()) {
-    statusesContainer.removeChild(statusesContainer.firstChild);
-  }
-  statusesContainer.style.visibility = 'hidden';
+  document.getElementById("loading").style.visibility = 'visible';
```

Comment:
Why do you want to show the loading indicator when resetting the alerts? Is the loading indicator something you want to show by default? It would be better to show and hide the loading element only when you are actually loading something. Else your page or logic might end up in a state that could be confusing to the user.

Reply:
On second thought, there doesn't seem to be a reason to reset user alerts, since the only way to reload recommendations (as of now) would be to refresh the page, which gets handled with default values. I removed this for now, but will consider adding a "try again?" button as you suggested. The google maps API doesn't specify if results will always be different if the same query is passed in, but from some past experimentation the results seem to always be the same.

Updated